src/components/Home/WelcomeHeader.js

```diff
@@ -37,6 +48,94 @@ const WelcomeHeader = () => {
       },
     },
   };
+  
+  useEffect(() => {
+    if (localPath !== ''){
+      const installSampleProtocol = async() => {
+        return openNetcanvas(localPath)
+      }
+      installSampleProtocol()
+    }
+  }, [localPath])
+
+  const getURL = uri =>
+  new Promise((resolve, reject) => {
+    try {
+      resolve(new URL(uri));
+    } catch (error) {
+      reject(error);
+    }
+  });
+
+const urlError = friendlyErrorMessage("The location you gave us doesn't seem to be a valid URL. Check the location, and try again.");
+const networkError = friendlyErrorMessage("We weren't able to fetch your protocol. Your device may not have an active network connection, or you may have mistyped the URL. Ensure you are connected to a network, double check your URL, and try again.");
+const fileError = friendlyErrorMessage('The protocol could not be saved to your device. You might not have enough storage available. ');
+
+/**
+ * Download a protocol from a remote server.
+ *
+ * If the URL points to an instance of a Network Canvas Server, then the caller must ensure
+ * that the SSL certificate has been trusted. See {@link ApiClient#addTrustedCert}.
+ *
+ * @param {string} uri
+ * @return {string} output filepath
+ */
+const downloadProtocolFromURI = inEnvironment((environment)=> {
+
+  if (environment === environments.ELECTRON) {
+    const request = require('request-promise-native');
+    const path = require('path');
+    const electron = require('electron');
+    const { dialog } = electron.remote
+    const tempPath = (electron.app || electron.remote.app).getPath('temp');
+    const from = path.join(tempPath, 'SampleProtocol') + '.netcanvas';
+    const fs = require('fs')
+
+    const selectPath = new Promise(function(resolve, reject){
+      dialog.showOpenDialog(null, {
+        properties: ['openDirectory']
+      })
+      .then((result) => {
+        if (result.filePaths.toString() !== ''){
+          const destination = path.join(result.filePaths.toString(), 'SampleProtocol') + '.netcanvas';
+          resolve(destination);
+        }
+      })
+    })
```

Comment:
You create a problem for yourself here by using an open dialog rather than a save dialog.

Because you use an open dialog, you only get a path bath with no filename associated with it. This means that you once again incorrectly need to hard code the filename into the path. 

If you use a save dialog, you will also get a filename that the user has chosen. You can pre-populate this dialog with the actual sample protocol name, based on the URL.